Error with serve in example 'with-firebase-hosting'

I got an error when I use the command `npm run serve`:
```
⚠  functions: Failed to load functions source code. Ensure that you have the latest SDK by running npm i --save firebase-functions inside the functions directory.
⚠  functions: Error from emulator. Error parsing triggers: Cannot find module '/Users/arthur/Documents/with-firebase-hosting/dist/functions'

Try running "npm install" in your functions directory before deploying.
```

To make it work I have to add `preserve` and `install-deps` scripts in `package.json` (as seen [on jthegedus example](https://github.com/jthegedus/firebase-functions-next-example/blob/1b9803ac2ec4366435d093aacfddc6d7a1b15206/package.json))

Maybe it should be added to the example ?

@jthegedus Thanks !

--- a/examples/with-firebase-hosting/firebase.json
+++ b/examples/with-firebase-hosting/firebase.json
@@ -6,11 +6,9 @@
         "source": "**/**",
         "function": "next"
       }
-    ],
-    "predeploy": "npm run build-public"
+    ]
   },
   "functions": {
-    "source": "dist/functions",
-    "predeploy": "npm run build-funcs && npm run build-app && npm run copy-deps"
+    "source": "dist/functions"
   }
 }

--- a/examples/with-firebase-hosting/package.json
+++ b/examples/with-firebase-hosting/package.json
@@ -1,26 +1,33 @@
 {
   "name": "with-firebase-hosting",
-  "version": "4.0.0",
-  "description": "Host Next.js SSR app on Firebase Cloud Functions with Firebase Hosting redirects.",
+  "version": "4.0.1",
+  "description":
+    "Host Next.js SSR app on Firebase Cloud Functions with Firebase Hosting redirects.",
   "scripts": {
-    "dev": "next \"src/app\"",
+    "dev": "next \"src/app/\"",
+    "preserve":
+      "npm run build-public && npm run build-funcs && npm run build-app && npm run copy-deps && npm run install-deps",
     "serve": "NODE_ENV=production firebase serve",
+    "predeploy":
+      "npm run build-public && npm run build-funcs && npm run build-app && npm run copy-deps",
     "deploy": "firebase deploy",
     "clean": "rimraf \"dist/functions/**\" && rimraf \"dist/public\"",
     "build-public": "cpx \"src/public/**/*.*\" \"dist/public\" -C",
     "build-funcs": "babel \"src/functions\" --out-dir \"dist/functions\"",
-    "build-app": "next build \"src/app\"",
-    "copy-deps": "cpx \"*{package.json,package-lock.json,yarn.lock}\" \"dist/functions\" -C"
+    "build-app": "next build \"src/app/\"",
+    "copy-deps":
+      "cpx \"*{package.json,package-lock.json,yarn.lock}\" \"dist/functions\" -C",
+    "install-deps": "cd \"dist/functions\" && npm i"
   },
   "dependencies": {
-    "firebase-admin": "5.12.1",
-    "firebase-functions": "1.0.2",
-    "next": "latest",
-    "react": "16.3.2",
-    "react-dom": "16.3.2"
+    "firebase-admin": "^5.12.1",
+    "firebase-functions": "^1.0.2",
+    "next": "^6.0.3",
+    "react": "^16.3.2",
+    "react-dom": "^16.3.2"
   },
   "devDependencies": {
-    "@babel/cli": "7.0.0-beta.42",
+    "@babel/cli": "^7.0.0-beta.42",
     "cpx": "1.5.0",
     "firebase-tools": "3.18.4",
     "prettier": "1.12.1",

--- a/examples/with-firebase-hosting/src/app/.babelrc
+++ b/examples/with-firebase-hosting/src/app/.babelrc
@@ -1,0 +1,3 @@
+{
+  "presets": [["next/babel"]]
+}

